qa/tasks/caasp.py

```diff
@@ -0,0 +1,86 @@
+'''
+Task that deploys a CAASP cluster on all the nodes
+Linter:
+    flake8 --max-line-length=100
+'''
+import logging
+from teuthology.misc import (
+    all_roles_of_type,
+    append_lines_to_file,
+    get_file,
+    )
+from teuthology.task import Task
+from tasks.util import get_remote_for_role
+log = logging.getLogger(__name__)
+
+
+class Caasp(Task):
+    """
+    Deploy a Caasp cluster on all remotes using Skuba.
+    Nodes are declared in suites/caasp folder
+    """
+
+    def __init__(self, ctx, config):
+        super(Caasp, self).__init__(ctx, config)
+        log.debug("beginning of constructor method")
+        self.ctx['roles'] = self.ctx.config['roles']
+        self.log = log
+        self.remotes = self.cluster.remotes
+        self.mgmt = get_remote_for_role(self.ctx, "skuba_mgmt_host.0")
+        self.ssh_priv = 'caasp_key.rsa'
+        self.ssh_pub = 'caasp_key.rsa.pub'
+        self.set_agent = "eval $(ssh-agent) && ssh-add ~/{} && ".format(
+            self.ssh_priv)
```

Comment:
(suggestion) I think, `set_agent` is better to rename to something like `set_agent_cmd` to be self-descriptive when referring later in the code, also `&&` is extra here, since, it stricts the usage of the command. For example, you can later use it by your choice:
```
remote.sh(self.set_agent_cmd + " && " + "ssh {node} cat /etc/os-release".format(remote.hostname))
```
or
```
remote.sh("""#!/bin/bash
{set_agent}
ssh {node} cat /etc/os-release
""".format(set_agent=self.set_agent_cmd, node=remote.hostname))
```
In other words it gives you a flexibility to control usage of this peace of code, and you can select '&&' or ';' or '\n' as command delimiter.

Reply:
This is cosmetic I think, and the code is working.
We can update if that is needed later one.